XenAdmin/Controls/SrPickerItem.cs

```diff
@@ -269,17 +276,18 @@ private void Update()
             Text = TheSR.Name();
             SetImage();
 
-            if (UnsupportedSR)
+            if (UnsupportedSR || !TheSR.SupportsVdiCreate() ||
+                !TheSR.Show(Properties.Settings.Default.ShowHiddenVMs))
                 return;
 
-            if (ShowHiddenVDIs && !ExistingVDILocation() && CanBeEnabled)
+            if (CanBeEnabled)
             {
                 Description = string.Format(Messages.SRPICKER_DISK_FREE, Util.DiskSizeString(TheSR.FreeSpace(), 2),
                     Util.DiskSizeString(TheSR.physical_size, 2));
                 Enabled = true;
                 Show = true;
             }
-            else if (TheSR.PBDs.Count > 0 && TheSR.SupportsVdiCreate())
+            else

```

Comment:
This has the effect of showing a detached SR (no PBDs) in the list (correctly disabled), while previously detached SRs were not displayed. But I think this is OK

Reply:
Hmm, I've tried to simplify things, but the logic is still confusing. The properties in the SrPickerItem subclasses check for IsDetached() and also provide the disabled reason, so they give the imperssion that it's ok to show them. On the other hand, the same properties also check for SupportsVdiCeate(), which is not needed since we hide them here, but I was nervous about  removing it. Should I just move all the checks into the subclasses and merge the supported and CanBeEnabled logic into one tri-state property (returning enabe|disable|hide)?

Reply:
We could do the sugested improvement later, for now I am happy with these changes